### ISSUE TITLE
upload in public shared folder creates errors 

### ISSUE THREAD

**Issue by CaptionF** [owncloud/core#18110]
Steps to reproduce

    create directory in owncloud
    share this directory via public link and "allow editing"
    copy the link, open another browser and upload file

Expected behaviour

file should be uploaded and no error should be seen
Actual behaviour

when uploaded files into the folder through the public link share, there are always errors reported:

extract from the log file
```
Aug 6 13:41:40 frdv01676 ownCloud[3662]: {hook} error while running hook (OCA\Activity\HooksStatic::fileCreate): An exception occurred while executing 'INSERT INTO "oc_activity"("app", "subject", "subjectparams", "message", "messageparams", "file", "link", "user", "affecteduser", "timestamp", "priority", "type") VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? )' with params ["files", "created_by", "a:2:{i:0;s:28:\"\/Photos\/test\/hochebene-2.jpg\";i:1;s:0:\"\";}", "", "a:0:{}", "\/Photos\/test\/hochebene-2.jpg", "***\/index.php\/apps\/files?dir=%2FPhotos%2Ftest", false, "***", 1438861300, 40, "file_created"]
ORA-01400: cannot insert NULL into ("OWNCLOUD"."oc_activity"."user")
```

The upload has worked fine, but there error seems strange. Of course, due to public upload, there is no user name.
Server configuration

Operating system:
SLES12

Web server:
Apache 2.4
Database:
Oracle
PHP version:
5.5.14
ownCloud version: (see ownCloud admin page)
oc8.0.5
Updated from an older ownCloud or fresh install:
Instance to be validated: some time ago from 7.0.5 to 8.0.3.

**Comment by CaptionF** [comment]
https://github.com/owncloud/enterprise/issues/752

**Comment by PVince81** [comment]
I couldn't reproduce this locally, neither with 8.0.3 nor 8.0.5.
But then I saw "ORA" in the error message, which is an Oracle error message.

Maybe the SQL query needs special handling for Oracle...

@nickvergessen @DeepDiver1975

**Comment by nickvergessen** [comment]
@CaptionF fixed for 8.1.2 and 8.0.7
